dask/array/creation.py

```diff
@@ -668,10 +678,10 @@ def diag(v, k=0):
             for j, n in enumerate(chunks_1d):
                 key = (name, i, j)
                 if i == j:
-                    dsk[key] = (np.diag, blocks[i])
+                    dsk[key] = Task(key, np.diag, TaskRef(blocks[i]))
                 else:
-                    dsk[key] = (np.zeros, (m, n))
-                    dsk[key] = (partial(np.zeros_like, shape=(m, n)), meta)
+                    # dsk[key] = Task(key, np.zeros, (m, n))
+                    dsk[key] = Task(key, partial(np.zeros_like, shape=(m, n)), meta)
```

Comment:
FYI No need for partials (but it shouldn't be harmful). `Task` accepts kwargs as well
```suggestion
                    dsk[key] = Task(key, np.zeros_like, meta, shape=(m, n))
```